gitgud/operations.py

```diff
@@ -159,51 +159,53 @@ def commit(self, commit_message, parents, time_offset):
 
     def file_in_commit(self, commit, filepath):
         commit = self.repo.commit(commit)
+        if not isinstance(filepath, str):
+            filepath = str(filepath)
         return filepath in commit.tree
 
-    def get_commit_content(self, commit, filepath):
-        if commit in self._streamed_content:
-            return self._streamed_content[commit]
-
+    def get_commit_file_content(self, commit, filepath):
         commit = self.repo.commit(commit)
-
         if not isinstance(filepath, str):
             filepath = str(filepath)
 
-        commit_content = (commit.tree / filepath).data_stream.read().decode("ascii")  # noqa: E501
-        self._streamed_content[commit] = commit_content
-        return commit_content
-
-    def get_staging_area(self):
-        return self.get_tracked_changes("staging")
+        if commit in self._streamed_content and \
+                filepath in self._streamed_content[commit]:
+            return self._streamed_content[commit][filepath]
 
-    def get_working_area(self):
-        changes = self.get_tracked_changes("working")
-        changes["added"] += self.repo.untracked_files
-        return changes
+        commit_content = (commit.tree / filepath) \
+            .data_stream.read().decode("ascii")
+        if commit not in self._streamed_content:
+            self._streamed_content[commit] = {}
 
-    def get_tracked_changes(self, change_type):
-        if change_type == "staging":
-            staged_files = self.repo.index.diff("HEAD")
-        elif change_type == "working":
-            staged_files = self.repo.index.diff(None)
-        else:
-            raise KeyError
+        self._streamed_content[commit][filepath] = commit_content
+        return commit_content
 
-        change_key_map = {
-            "added": "A",
-            "deleted": "D",
-            "modified": "M",
-            "renamed": "R"
-        }
+    def get_commit_content(self, commit):
+        content = {}
+        for filepath in commit.tree:
+            content.update(
+                {filepath: self.get_commit_file_content(commit, filepath)}
+            )
+        return content
```

Comment:
This does not get the content of sub-trees (directories).

If I have a subdirectory, I can do something like this:
```
>>> repo.head.commit.tree.trees[0]
<git.Tree "1b7d3913a8f0d539997a1d7d21ae6181061fc7f9">
>>> repo.head.commit.tree.trees[0].blobs
[<git.Blob "c47b355949ca027c0f929e24343279458228a04f">]
```
[Trees](https://gitpython.readthedocs.io/en/stable/reference.html?highlight=to_blob#git.objects.tree.Tree) can have subtrees and we should handle that.

Reply:
`repo.head.commit.tree.traverse()` should do the trick. Be aware that this also returns trees, which you would likely want to skip.